Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -3703,16 +3797,62 @@ def visit_GILExitNode(self, node):
         self.visitchildren(node)
         return node
 
-    def visit_Node(self, node):
+    def visit_Node(self, node, coerce_to_temp_node=False):
         if self.env_stack and self.nogil and node.nogil_check:
             node.nogil_check(self.env_stack[-1])
+        if not coerce_to_temp_node:
+            self.was_coerce_to_temp = False
```

Comment:
Can't this be done outside of the method? The `coerce_to_temp_node` option seems unnecessary.